### PR TITLE
Adds wield timer for guns

### PR DESCRIPTION
**Pull request by vode-code**
## About The Pull Request
This PR makes it so larger guns take extra time to wield, with the exception of the AMR as its reload time is already bad and it requires two hands to reload.

## Why It's Good For The Game
there is currently very little downside to using bigger guns, and this changes that.

## Changelog
:cl: Chickenish
balance: guns larger than a pistol now take time to wield, with the exception of the ballistic sniper.
/:cl:

### CODE REVIEW

**Comment by ValoTheValo** [comment]
3 second wield time for SMG is idiotic. this will just make combat more tedious and makes the game worse. nobody asked for this

**Comment by SirRichardFrancis** [comment]
ugomonis

**Comment by vode-code** [comment]
numbers have been changed, thoughts?

**Review by SirRichardFrancis — COMMENTED** [review]
Can't find where `wield_delay` used for checks. Did you test that code?

**Comment by vode-code** [comment]
> 
> 
> Can't find where `wield_delay` used for checks. Did you test that code?

was specifically working on the numbers, have made and tested the core and fixed the spaces now.

**Comment by vode-code** [comment]
> 
> 
> Conflicts with #6584, by the way.

shotgun/pump/grenade is... odd.

File: code/modules/projectiles/guns/projectile/automatic/type17.dm [hunk 1]
```diff
@@ -31,6 +31,7 @@
     spawn_blacklisted = TRUE //until loot rework
 
     gun_tags = list(GUN_SILENCABLE)
+	wield_delay = 3 //deciseconds
```
[the remark this answers is not in the document]
Comment (zombiemimic) [reply code/modules/projectiles/guns/projectile/automatic/type17.dm:34]: good heavens, what is this indentation?

**Comment by SirRichardFrancis** [comment]
> shotgun/pump/grenade is... odd.

Because procs needed for grenade launcher is there. Reinventing stuff is bad, m'okay.

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 2]
```diff
@@ -24,6 +24,7 @@
 		list(mode_name="Melt", mode_desc="A much more potent plasma round for breaching tough opponents' hides", projectile_type=/obj/item/projectile/plasma, fire_sound='sound/weapons/Laser.ogg', fire_delay=12, charge_cost=25, icon="kill", projectile_color = "#FF0000"),
 		list(mode_name="INCINERATE", mode_desc="An armor-stripping plasma round", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=14, charge_cost=30, icon="destroy", projectile_color = "#FFFFFF"),
 	)
+	wield_delay = 0.3
```
Comment (zombiemimic) [inline code/modules/projectiles/guns/energy/plasma.dm:27]: what unit is this in?

Comment (SirRichardFrancis) [reply code/modules/projectiles/guns/energy/plasma.dm:27]: Centiseconds?
image

Comment (zombiemimic) [reply code/modules/projectiles/guns/energy/plasma.dm:27]: fastest plasma rifle in the west

Comment (vode-code) [reply code/modules/projectiles/guns/energy/plasma.dm:27]: thanks for the catch, I meant for three deciseconds.

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -845,3 +846,9 @@
 		H.using_scope = src
 	else
 		H.using_scope = null
+
+obj/item/gun/wield(user)
```
Comment (MLGTASTICa) [inline code/modules/projectiles/gun.dm:850]: add a / to the very start of the proc , inconsistent with all the other overrides in the folder

File: code/modules/projectiles/guns/projectile/automatic/type17.dm [hunk 4]
```diff
@@ -1,36 +1,37 @@
...
+	unload_sound = 'sound/weapons/guns/interact/ltrifle_magout.ogg'
+	reload_sound = 'sound/weapons/guns/interact/ltrifle_magin.ogg'
+	cocked_sound = 'sound/weapons/guns/interact/ltrifle_cock.ogg'
+	zoom_factor = 0.6
+	recoil_buildup = 1.5
+	one_hand_penalty = 15 //automatic rifle level
+	damage_multiplier = 1.3
+	penetration_multiplier = 1.7
+	spawn_tags = SPAWN_TAG_GUN_OS
+	init_firemodes = list(
```
Comment (MLGTASTICa) [inline code/modules/projectiles/guns/projectile/automatic/type17.dm:27]: Fix all this indentation jesus christ.

Comment (MLGTASTICa) [reply code/modules/projectiles/guns/projectile/automatic/type17.dm:27]: Nevermind , github shows it as spaces but its indents.

Comment (vode-code) [reply code/modules/projectiles/guns/projectile/automatic/type17.dm:27]: that IS me fixing the indentation.

File: code/modules/projectiles/guns/projectile/automatic/vintorez.dm [hunk 5]
```diff
@@ -25,6 +25,7 @@
 		SEMI_AUTO_NODELAY,
 		FULL_AUTO_400
 		)
+	wield_delay = 3 //deciseconds
```
Comment (MLGTASTICa) [inline code/modules/projectiles/guns/projectile/automatic/vintorez.dm:28]: full blown assault rifle , make it 6 deciseconds

File: code/modules/projectiles/guns/projectile/automatic/wintermute.dm [hunk 6]
```diff
@@ -34,6 +34,7 @@
 	spawn_tags = SPAWN_TAG_FS_PROJECTILE
 
 	gun_tags = list(GUN_SILENCABLE)
+	wield_delay = 3 //deciseconds
```
Comment (MLGTASTICa) [inline code/modules/projectiles/guns/projectile/automatic/wintermute.dm:37]: full blown assault rifle , make it 6 deciseconds please

File: code/modules/projectiles/guns/projectile/automatic/z8.dm [hunk 7]
```diff
@@ -34,6 +34,7 @@
 	var/obj/item/gun/launcher/grenade/underslung/launcher
 
 	spawn_tags = SPAWN_TAG_FS_PROJECTILE
+	wield_delay = 3 //deciseconds
```
Comment (MLGTASTICa) [inline code/modules/projectiles/guns/projectile/automatic/z8.dm:37]: full blown rifle , please make it 6 deciseconds

File: code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm [hunk 8]
```diff
@@ -32,6 +32,7 @@
 	sawn = /obj/item/gun/projectile/boltgun/obrez
 	var/bolt_open = 0
 	var/item_suffix = ""
+	wield_delay = 8 //deciseconds
```
Comment (MLGTASTICa) [inline code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm:35]: its sawed off , why 8 deciseconds ? it should be only 4 or 3

**Review by MLGTASTICa — APPROVED** [review]

File: code/modules/projectiles/guns/projectile/automatic/vintorez.dm (hunk 5 above)
Comment (ValoTheValo) [reply code/modules/projectiles/guns/projectile/automatic/vintorez.dm:28]: he went off my table, goes off caliber .20 should be like .4 or .5 iirc

Comment (vode-code) [reply code/modules/projectiles/guns/projectile/automatic/vintorez.dm]: done

File: code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm (hunk 8 above)
Comment (vode-code) [reply code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm:35]: done

**Comment by Alliostra** [comment]
Not sure if this is a good idea, wielding long arms already takes some time to do because it requires one more manual input than drawing a gun. Plus, you can also dual-wield most pistols for devastating effect, dual avaraslas are no joke.

**Comment by drexample** [comment]
Closing because https://github.com/discordia-space/CEV-Eris/pull/6713

**Pull request closed without merging**